pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -615,6 +615,18 @@ func (proxyStat *ProxyRequestContext) IsTimeout() bool {
 	return false
 }
 
+// DNSProxyConfig struct
```

Comment:
nit: let's a a slightly more verbose godoc comment here, e.g.

```suggestion
// DNSProxyConfig is the configuration for the DNS proxy.
```